### PR TITLE
Add an invite code field to the register page

### PR DESCRIPTION
**Pull request by retcinder**
## TODO
- [x] Make it actually submit the invite code
- [x] Add env var

## In Action
image

### CODE REVIEW

**Comment by retcinder** [comment]
@amycatgirl or @Rexogamer, mind checking this out when you're ready?

File: packages/client/components/auth/src/flows/Form.tsx [hunk 1]
```diff
@@ -72,6 +80,8 @@ export function Fields(props: FieldProps) {
   const fieldConfiguration = useFieldConfiguration();
   const [failedValidation, setFailedValidation] = createSignal(false);
 
+  const inviteCodeNeeded: boolean|undefined = clientController.lifecycle.client.configuration?.features.invite_only;
```
Comment (catuhana) [inline packages/client/components/auth/src/flows/Form.tsx]: A simple missed formatting:
```suggestion
  const inviteCodeNeeded: boolean | undefined = clientController.lifecycle.client.configuration?.features.invite_only;
```

File: packages/client/components/auth/src/flows/Form.tsx [hunk 2]
```diff
@@ -81,7 +91,7 @@ export function Fields(props: FieldProps) {
 
   return (
     <For each={props.fields}>
-      {(field) => (
+      {(field) => (field != "invite" || inviteCodeNeeded) && (
```
Comment (catuhana) [inline packages/client/components/auth/src/flows/Form.tsx]: It's better to use https://docs.solidjs.com/concepts/control-flow/conditional-rendering#show here? SolidJS recommends their components for better performance and such.

Comment (retcinder) [reply packages/client/components/auth/src/flows/Form.tsx]: Will adapt to that recommendation later

**Comment by amycatgirl** [comment]
I'll give it a look tonight, I'm going to be away from home for a couple of hours

File: packages/client/components/auth/src/flows/Form.tsx [hunk 3]
```diff
@@ -82,30 +92,32 @@ export function Fields(props: FieldProps) {
...
+                  {...fieldConfiguration[field]}
+                  name={field}
+                  placeholder={fieldConfiguration[field].placeholder()}
+                  submissionTried={failedValidation()}
+                  onInvalid={onInvalid}
+                />
+              </>
+            )}
+          </FormGroup>
+        </Show>
```
Comment (amycatgirl) [inline packages/client/components/auth/src/flows/Form.tsx]: Nit: I would rather use a Switch component here, rather than using ternaries.

**Review by amycatgirl — APPROVED** [review]
Other than that nit comment I left you, the code looks good to me.

File: packages/client/components/auth/src/flows/Form.tsx (hunk 3 above)
Comment (retcinder) [reply packages/client/components/auth/src/flows/Form.tsx]: Do you mean the bit inside the FormGroup? That hasn't been changed (indentation was added) but I can change it up to use that

Comment (amycatgirl) [reply packages/client/components/auth/src/flows/Form.tsx]: Oh i see

I was reviewing the pr from mobile so it wasn't clear what exactly was changed

you can leave it as is for now

Comment (retcinder) [reply packages/client/components/auth/src/flows/Form.tsx]: I just changed it though :weh:

Comment (amycatgirl) [reply packages/client/components/auth/src/flows/Form.tsx]: eh it's fine
it should do the same thing anyways :trollface:

Comment (retcinder) [reply packages/client/components/auth/src/flows/Form.tsx]: ok but is the new code good? (I'm not good with JS so I need this to be double checked) @amycatgirl

**Review by amycatgirl — APPROVED** [review]

File: packages/client/components/auth/src/flows/Form.tsx (hunk 3 above)
Comment (amycatgirl) [reply packages/client/components/auth/src/flows/Form.tsx]: LGTM

**Comment by retcinder** [comment]
Sophie approved in my DMs, merging

**Pull request merged**